server/processors/src/test/java/org/hawkular/apm/server/processor/nodedetails/NodeDetailsDeriverTest.java

```diff
@@ -170,4 +170,67 @@ public void testInitial() throws RetryAttemptException {
         assertFalse(details.get(1).isInitial());
     }
 
+    @Test
+    public void testCalculateActualTimeSync() throws RetryAttemptException {
```

Comment:
> testCalculateActualTimeAsync I believe satisfies the second and third cases?

Yes, I'm thinking about this as white box testing. The second point is not satisfied:  in ` testCalculateActualTimeAsync` sum of elapsed times greater than parent's duration.